plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs

```diff
@@ -521,6 +526,46 @@ instance (HasConstantIn uni term, GShow uni, GEq uni, uni `Contains` a) =>
 -- | A constraint for \"@a@ is a 'KnownType' by means of being included in @UniOf term@\".
 type KnownBuiltinType term a = KnownBuiltinTypeIn (UniOf term) term a
 
+{- Note [Performance of KnownTypeIn instances]
+Even though we don't use 'makeKnown' and 'readKnown' directly over concrete types, it's still
+beneficial to inline them, because otherwise GHC compiles each of them to two definitions
+(one calling the other) for some reason.
```

Comment:
I'm guessing there's no way to tell the compiler that every implementation of a method should be inlined, since that's being done manually for each one, which is a bit annoying. Presumably the compiler would have to be able to find pragmas in other files if it wanted to do that.  Maybe the comment could be more forceful about telling people to make sure they add `INLINE`.